nipoppy_cli/docs/source/user_guide/tracking.md

```diff
@@ -0,0 +1,130 @@
+# Tracking pipeline processing status
+
+Nipoppy trackers search for expected file paths or patterns in pipeline output files. They are specific to pipelines and versions, and can be configured to include custom paths for various levels of granularity.
+
+## Summary
+
+### Prerequisites
+
+- A Nipoppy dataset with a valid global configuration file and an accurate manifest
+    - See the [Quickstart guide](../quickstart.md) for instructions on how to set up a new dataset
+- Processed imaging data the {{dpath_derivatives}} directory
+    - See <project:processing.md> for expected subdirectory structure
+
+### Data directories and files
+
+| Directory or file | Content description |
+|---|---|
+| {{dpath_derivatives}} | **Input** -- {{content_dpath_derivatives}} |
+| {{fpath_imaging_bagel}} | **Output** -- Tabular file containing processing status for each participant/session and pipeline |
+
+### Commands
+
+- Command-line interface: [`nipoppy track`](<project:../cli_reference/track.md>)
+- Python API: {class}`nipoppy.workflows.PipelineTracker`
+
+### Workflow
+
+1. Nipoppy will loop over all participants/sessions that have BIDS data according to the {term}`doughnut file`
+2. For each participant-session pair:
+    1. Paths in the pipeline's tracker configuration will be processed such that template strings related to the participant/session are replaced by the appropriate values
+    2. Each path in the list is checked, then a status is assigned, and the bagel file is updated accordingly
+
+## Configuring a pipeline tracker
+
+The global configuration file should include paths to tracker configuration files, which are {term}`JSON` files containing lists of dictionaries.
+
+Here is example of tracker configuration file (default for MRIQC 23.1.0):
+```{literalinclude} ../../../nipoppy/data/examples/sample_tracker_configs/mriqc-23.1.0.json
+```
+
+```{tip}
+- The paths are expected to be relative to the {{dpath_derivatives}}`/<PIPELINE_NAME>/<PIPELINE_VERSION>/output` directory.
+- "Glob" expressions (i.e., that include `*`) are allowed in paths.
+```
+
+```{note}
+The template strings `[[NIPOPPY_<ATTRIBUTE_NAME>]]` work the same way as the ones in the [global configuration file](<global-config-template-strings>) and the [pipeline invocation files](<invocation-template-strings>) -- they are replaced at runtime by appropriate values.
+```
+
+```{attention}
+Currently, only the tracker configuration with `pipeline_complete` in the `NAME` field will be used, but we are planning to extend trackers to allow multiple statuses per pipeline. Stay tuned!
```

Comment:
We should also note couple of things:
1. the default `pipeline_complete` criteria is quite minimal and does not check all output files generated from a default pipeline run. 
2. We do not rely on the exit codes.